lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -944,10 +961,29 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
              cast<ValueDecl>(D)->isInstanceMember()))
           continue;
         
-        if (isa<FuncDecl>(candidate) &&
-            (cast<FuncDecl>(candidate)->getParameters()->size() !=
-             cast<FuncDecl>(D)->getParameters()->size()))
-          continue;
+        if (isa<FuncDecl>(candidate)) {
+          auto cParams = cast<FuncDecl>(candidate)->getParameters();
+          auto dParams = cast<FuncDecl>(D)->getParameters();
+          
+          if (cParams->size() != dParams->size())
+            continue;
+          
+          bool hasSameParameterTypes = true;
+          for (auto index : indices(*cParams)) {
+            TypeMatchOptions matchMode = TypeMatchFlags::AllowOverride;
+            matchMode |= TypeMatchFlags::AllowTopLevelOptionalMismatch;
```

Comment:
If we're going to use these flags here, we need tests for them, i.e. where the parameter types don't match exactly but do follow the override rules.

Reply:
(Alternately, you could take this back out, since we only decided to add it late in the game.)